server/core/src/rodsServer.cpp

```diff
@@ -305,6 +313,130 @@ namespace
 
         return 0;
     } // create_pid_file
+
+    void create_stacktrace_directory()
+    {
+        namespace fs = boost::filesystem;
+        boost::system::error_code ec;
+        fs::create_directories("../.." / fs::path{IRODS_DEFAULT_PATH_HOMEDIR} / "stacktrace", ec);
+    } // create_stacktrace_directory
+
+    void log_stacktrace_files(std::atomic<bool>& _stop_token)
+    {
+        namespace fs = boost::filesystem;
+
+        // The leading "../.." enables support for non-package installations.
+        // In a packaged installation, this will produce "/var/lib/irods/stacktrace".
+        const auto stacktrace_directory = "../.." / fs::path{IRODS_DEFAULT_PATH_HOMEDIR} / "stacktrace";
+
+        // Get the sleep time.
+        const auto sleep_time = [] {
+            try {
+                const auto adv_settings = irods::server_properties::instance().map()
+                    .at(irods::CFG_ADVANCED_SETTINGS_KW);
+
+                const auto iter = adv_settings
+                    .find(irods::CFG_STACKTRACE_FILE_PROCESSOR_SLEEP_TIME_IN_SECONDS_KW);
+
+                if (iter != std::end(adv_settings)) {
+                    if (const auto seconds = iter->get<int>(); seconds > 0) {
+                        return seconds;
+                    }
+                }
+            }
+            catch (...) {}
+
+            return 3; // The default sleep time.
+        }();
+
+        using log = irods::experimental::log;
+
+        while (!_stop_token.load()) {
+            for (auto&& entry : fs::directory_iterator{stacktrace_directory}) {
+                // Expected filename format:
+                //
+                //     <epoch_seconds>.<remaining_milliseconds>.<agent_pid>
```

Comment:
I might have missed it... what is meant by "remaining" in `remaining_milliseconds` here?

Reply:
it represents partial seconds. so it provides more accuracy as to when the stacktrace happened.

see `struct timespec` in https://man7.org/linux/man-pages/man2/clock_getres.2.html

Reply:
I guess I was more wondering if it wouldn't make more sense to say `epoch_milliseconds` or somesuch. The use of the word "remaining" was mildly confusing for me. It's fine as is :)

Reply:
changing to `epoch_milliseconds`.

Reply:
done